src/cache/cache/green_cache.cpp

```diff
@@ -0,0 +1,181 @@
+// Rafael Sá Menezes - 04/2020
+
+#include "green_cache.h"
+#include <cache/algorithms/algorithms.h>
+#include <cache/algorithms/algorithms_utils.h>
+
+void green_cache::canonize_expr(expr2tc &expr)
+{
+  expr_variable_reordering reordering(expr);
+  reordering.run();
+
+  expr_green_normal_form gnf(expr);
+  gnf.run();
+
+  // TODO: add variable renaming
+}
+
+void green_cache::run_on_assert(symex_target_equationt::SSA_stept &step)
```

Comment:
Add short description of all methods.

Reply:
The header includes a description of the method. In this case, the mother class has the description